Groups/GroupAttendanceMulti.ascx.cs

```diff
@@ -0,0 +1,515 @@
+        {
+            var displayGroupName = GetAttributeValue( AttributeKey.DisplayGroupNames ).AsBoolean();
+            if ( displayGroupName )
+            {
+                lHeading.Text = $"{BlockName} - {_groups.Select( g => g.Name ).JoinStringsWithCommaAnd()}";
+            }
+            else
+            {
+                lHeading.Text = BlockName;
+            }
+
+            if ( !_members.Any() )
+            {
+                nbNotice.Text = "There are no members to display for the selected group(s). Please check if you have permission to Manage Members or Edit the group(s) selected.";
+            }
+
+            _attendanceDate = PageParameter( PageParameterKey.Date ).AsDateTime() ?? PageParameter( PageParameterKey.Occurrence ).AsDateTime();
+
+            if ( _attendanceDate.HasValue )
+            {
+                lAttendanceDate.Text = _attendanceDate.ToShortDateString();
+                lAttendanceDate.Visible = true;
+                dpAttendanceDate.Visible = false;
+            }
+            else
+            {
+                dpAttendanceDate.SelectedDate = dpAttendanceDate.SelectedDate ?? RockDateTime.Now;
+                _attendanceDate = dpAttendanceDate.SelectedDate;
+                dpAttendanceDate.Visible = true;
+                lAttendanceDate.Visible = false;
+            }
+
+            lSchedule.Text = _groups.Where( g => g.Schedule != null ).ToList().Select( g => g.Schedule.FriendlyScheduleText ).JoinStrings( ", " );
+        }
+
+        /// <summary>
+        /// Binds the group members/attendees repeater.
+        /// </summary>
+        protected void BindRepeat()
+        {
+            var attendedIds = new List<int>();
+            using ( var rockContext = new RockContext() )
+            {
+                var groupIds = _groups.Select( g => g.Id ).ToList();
+                attendedIds = new AttendanceService( rockContext )
+                    .Queryable().AsNoTracking()
+                    .Where( a =>
+                        DbFunctions.DiffDays( a.StartDateTime, _attendanceDate ) == 0 &&
+                        a.DidAttend.HasValue &&
+                        a.DidAttend.Value &&
+                        a.Occurrence != null &&
+                        groupIds.Contains( a.Occurrence.GroupId.Value ) &&
+                        a.PersonAlias != null )
+                    .Select( a => a.PersonAlias.PersonId )
+                    .Distinct()
+                    .ToList();
+            }
+            var searchParts = tbSearch.Text.ToLower().SplitDelimitedValues();
+            _attendees = _members
+                .Where( gm => tbSearch.Text.IsNullOrWhiteSpace() ||
+                      ( searchParts.Length == 1 && gm.Person.LastName.ToLower().StartsWith( searchParts[0] ) ) ||
+                      ( searchParts.Length > 1 && gm.Person.FirstName.ToLower().StartsWith( searchParts[0] ) && gm.Person.LastName.ToLower().StartsWith( searchParts[1] ) )
```

Comment:
Need to add nickname to these queries as well.
```suggestion
                      ( searchParts.Length > 1 && gm.Person.FirstName.ToLower().StartsWith( searchParts[0] ) && gm.Person.LastName.ToLower().StartsWith( searchParts[1] ) ||
                      ( searchParts.Length > 1 && gm.Person.FirstName.ToLower().StartsWith( searchParts[1] ) && gm.Person.LastName.ToLower().StartsWith( searchParts[0] ) )
```